flatfs/flatfs.go

```diff
@@ -47,21 +46,19 @@ func New(path string, prefixLen int, sync bool) (*Datastore, error) {
 		return nil, ErrBadPrefixLen
 	}
 	fs := &Datastore{
-		path: path,
-		// convert from binary bytes to bytes of hex encoding
-		hexPrefixLen: prefixLen * hex.EncodedLen(1),
-		sync:         sync,
+		path:      path,
+		prefixLen: prefixLen,
+		sync:      sync,
 	}
 	return fs, nil
 }
 
-var padding = strings.Repeat("_", maxPrefixLen*hex.EncodedLen(1))
+var padding = strings.Repeat("_", maxPrefixLen)
 
 func (fs *Datastore) encode(key datastore.Key) (dir, file string) {
-	safe := hex.EncodeToString(key.Bytes()[1:])
```

Comment:
@whyrusleeping, note that the point of `key.Bytes()[1:]` this was to remove the '/' prefix'...